controllers/masteruserrecord/masteruserrecord_controller_test.go

```diff
@@ -234,36 +255,216 @@ func TestCreateMultipleUserAccountsSuccessful(t *testing.T) {
+			HasStatusUserAccounts(commontest.Member2ClusterName).
+			HasFinalizer()
+		AssertThatCountersAndMetrics(t).
+			HaveUsersPerActivationsAndDomain(toolchainv1alpha1.Metric{
+				"1,internal": 1, // unchanged
+			}).
+			HaveMasterUserRecordsPerDomain(toolchainv1alpha1.Metric{
+				string(metrics.Internal): 1, // unchanged
+			})
+
+		t.Run("the previous account should be deleted", func(t *testing.T) {
+			// when
+			result, err := cntrl.Reconcile(context.TODO(), newMurRequest(mur))
+			// then
+			require.NoError(t, err)
+			assert.Equal(t, 10*time.Second, result.RequeueAfter)
+			uatest.AssertThatUserAccount(t, "john", memberClient).
+				Exists().
+				MatchMasterUserRecord(mur).
+				HasLabelWithValue(toolchainv1alpha1.TierLabelKey, "deactivate30")
+			uatest.AssertThatUserAccount(t, "john", memberClient2).
+				DoesNotExist()
+			murtest.AssertThatMasterUserRecord(t, "john", hostClient).
+				HasConditions(toBeNotReady(toolchainv1alpha1.MasterUserRecordProvisioningReason, "")).
+				HasStatusUserAccounts(commontest.Member2ClusterName, commontest.MemberClusterName).
+				HasFinalizer()
+
+			t.Run("update the status when the UserAccount is gone", func(t *testing.T) {
+				// when
+				result, err := cntrl.Reconcile(context.TODO(), newMurRequest(mur))
+				// then
+				require.NoError(t, err)
+				assert.Empty(t, result.RequeueAfter)
+				uatest.AssertThatUserAccount(t, "john", memberClient).
+					Exists().
+					MatchMasterUserRecord(mur).
+					HasLabelWithValue(toolchainv1alpha1.TierLabelKey, "deactivate30")
+				uatest.AssertThatUserAccount(t, "john", memberClient2).
+					DoesNotExist()
+				murtest.AssertThatMasterUserRecord(t, "john", hostClient).
+					HasConditions(toBeNotReady(toolchainv1alpha1.MasterUserRecordProvisioningReason, "")).
+					HasStatusUserAccounts(commontest.MemberClusterName).
+					HasFinalizer()
+			})
 		})
+	})
+
+	t.Run("space that is being deleted should be ignored", func(t *testing.T) {
+		memberClient := commontest.NewFakeClient(t)
+		memberClient2 := commontest.NewFakeClient(t)
+		terminatingSpace := space.DeepCopy()
+		spacetest.WithDeletionTimestamp()(terminatingSpace)
+		spacetest.WithFinalizer()(terminatingSpace)
+
+		hostClient := commontest.NewFakeClient(t, mur, spaceBinding, terminatingSpace, sharedSpaceBinding, sharedSpace, toolchainStatus)
+		InitializeCounters(t, toolchainStatus)
+
+		cntrl := newController(hostClient, s, ClusterClient(commontest.MemberClusterName, memberClient), ClusterClient(commontest.Member2ClusterName, memberClient2))
+
+		// when reconciling
+		result, err := cntrl.Reconcile(context.TODO(), newMurRequest(mur))
+		// then
+		require.NoError(t, err)
+		assert.Empty(t, result.RequeueAfter)
+		uatest.AssertThatUserAccount(t, "john", memberClient).
+			DoesNotExist()
+		uatest.AssertThatUserAccount(t, "john", memberClient2).
+			DoesNotExist()
+		murtest.AssertThatMasterUserRecord(t, "john", hostClient).
+			HasConditions().
+			HasStatusUserAccounts().
+			HasFinalizer()
```

Comment:
> So the question is why the MUR should exist when it doesn't have access to any Spaces? 

Right that was the question.

>  MURs & Spaces are fully independent - they have different lifecycle. There is no automatic cleanup if MUR doesn't have access to any Space. It's also possible to create a MUR without any default Space - we do that a lot in e2e tests.

Thanks for explaining . That makes sense indeed!

Reply:
But the MUR with no space doesn't have any available functionality ATM in Sandbox right, or maybe there are some features that do not require a space/namespace ?

Reply:
we added this functionality because originally it was supposed to be used in AppStudio. Users should have been able to provision themself without any Space and then create the initial Space (if needed and the quota allows that) on demand later. 
Not sure if this is still a valid use case we need to support anymore. Dropping that would maybe simplify a few things in the code.

Reply:
got it! Thank for providing the background reason for this functionality! 

And indeed I guess dropping this functionality might allow us to remove some code. But since it's already there let's keep it.

Reply:
as discussed in the yesterday's call, we will drop it :-)

Reply:
Yeeeey 🥳 😄